samples/IdentityApp/IdentityApp/Program.fs

```diff
@@ -242,7 +242,10 @@ let configureServices (services : IServiceCollection) =
     services.AddCors() |> ignore
 
     // Configure Giraffe dependencies
-    services.AddGiraffe() |> ignore
+    services.AddGiraffe(
+        fun options ->
+            options.CompatibilityMode <- Version40
+        ) |> ignore
```

Comment:
@slang25 Is this what you had in mind? I think this is certainly more future proof and ASP.NET Core-ish, but I wonder if for Giraffe it wouldn't be better to have a `Func<IGiraffeOptions, IGiraffeOptions>` as a parameter intead of an `Action<IGiraffeOptions>`? It would be more functional to accept an object of `IGiraffeOptions` and then return a modified version of this object as part of the configuration instead of making direct modifications on the original `IGiraffeOptions` parameter. What do you think?

Reply:
e.g.:

```fsharp
services.AddGiraffe(fun options -> options.SetCompatibilityMode(Version40))
```

The `.SetCompatibilityMode` method would then return a new object with the mode being changed. This would also allow chaining of multiple settings if this object is going to grow in the future, e.g.:

```fsharp
services.AddGiraffe(
    fun options -> 
        options
            .SetCompatibilityMode(Version40)
            .SetSomethingElse(123)
            .SetYetAnotherValue(true)
)
```

Reply:
I'm away from my laptop to have a proper look, but this looks nice. Will have a proper look tonight.

Reply:
Also my PR (even though it got brutishly hijacked) got in for dotnet core 5 https://github.com/aspnet/Extensions/issues/1463 so we can actually make IOptions usable for records by then.

Reply:
I don't think that helps us here (it's a great change though), we aren't touching `IOptions` here.
I have generally found that records in public interface for things like options is great until you add a field, then every consuming `with` usage is broken, so any additions would be a major version change.

Reply:
@slang25 That's an interesting point about the use of record types!

Reply:
@slang25 wasn't claiming it was, just good to know of its existence. It's definitely useful for classes too, as it pertains to any type with non empty constructors.